sktime/forecasting/base/_adapters.py

```diff
@@ -12,6 +12,72 @@
 from sktime.forecasting.base._sktime import _SktimeForecaster
 from sktime.forecasting.base._sktime import _OptionalForecastingHorizonMixin
 
+from sktime.utils.validation.forecasting import check_y
+
+
+class _TbatsAdapter(_OptionalForecastingHorizonMixin, _SktimeForecaster):
+    """Base class for interfacing tbats forecasting algorithms"""
+
+    def fit(self, y, X=None, fh=None):
+        """Fit to training data.
+
+        Parameters
+        ----------
+        y : pd.Series
+            Target time series to which to fit the forecaster.
+        fh : int, list or np.array, optional (default=None)
+            The forecasters horizon with the steps ahead to to predict.
+        X : pd.DataFrame, optional (default=None)
+            Exogenous variables (ignored)
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        if X is not None:
+            raise NotImplementedError("BATS/TBATS don't support exog or endog data.")
```

Comment:
```suggestion
            raise NotImplementedError("BATS/TBATS don't support exogenous data.")
```

Reply:
Thinking about it, we probably shouldn't raise an error in this case. I see it this way: all forecasters accept X but some choose to ignore it. The problem is that this breaks composition workflows like ensembles where some forecasters may work with exogenous data but others don't. I believe that for the same reason, the `DummyClassifier` in scikit-learn also doesn't raise an error when passing X even though X is ignored. 

We could think about raising a warning. In that case, we should implement that in `check_y_X` with an additional keyword (something like `warn_if_exogenous`) and a corresponding unit tests based on estimator tags for `endogenous_only`. But not sure if that's necessary and probably best done in a separate PR.

Reply:
I have added a warning as you described in the code, but not yet in the tests.